### DISCUSSION TITLE
How to start it in linux (ubuntu arm64)

### DISCUSSION THREAD

**Discussion by jelebruh** [Pryaxis/TShock#2911] [Q&A]
I have a problem. 
If I start it with ./Tshock.Server 
```
You must install .NET to run this application. 
```
Im install it and I get other error (dotnet run Tshock.Server)
```
TShock.Server(1,1): error MSB4025: The project file could not be loaded. D
ata at the root level is invalid. Line 1, position 1.
```
In pages and faq I don't find any information about start on linux

**Comment by hakusaro** [comment]
Did you try without the `run`?

**Reply by jelebruh** [reply]
Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET program, but dotnet-TShock.Server does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.

**Comment by hakusaro** [comment]
Did you try ./Tshock.Server after you installed the .NET runtime?

**Reply by jelebruh** [reply]
> Did you try ./Tshock.Server after you installed the .NET runtime?

![Screenshot_2023-02-02-08-33-23-71_61c78dc80ee02b53007c815fefe993e3.jpg](https://user-images.githubusercontent.com/90205154/216240269-84f96925-13fb-43ee-a458-17da27705dcf.jpg)

**Comment by sgkoishi** [comment]
If you want to compile it yourself, run `dotnet run` inside the csproj folder, or `dotnet run --project TShockLauncher` inside the repo. https://github.com/Pryaxis/TShock/discussions/2903
Or, download the latest dev version. https://github.com/Pryaxis/TShock/discussions/2881#discussioncomment-4445111

**Reply by jelebruh** [reply]
I tried to build it. Dotnet send error cuz TerrariaAPI folder not found

**Reply by sgkoishi** [reply]
See [#2903](https://github.com/Pryaxis/TShock/discussions/2903#discussioncomment-4744066) for steps.

**Reply by jelebruh** [reply]
I get a... Too much errors like
` /home/servers/TShock/TShockAPI/Utils.cs(1162,32): error CS0246: The type or namespace name 'Vector2' could not be found (are you missing a using directive or an assembly reference?) [/home/servers/TShock/TShockAPI/TShockAPI.csproj]`

**Reply by sgkoishi** [reply]
Missing OTAPI, probably you should run `dotnet build` inside the repo first, or `dotnet run --project TShockLauncher`?

I would really suggest download the latest dev version if you don't know what to do

**Reply by jelebruh** [reply]
build not working 

TerrariaServerAPI.csproj" was not found. [/home/servers/TShock/TShock.sln]

I made this question because ANYTHING not working. ./binary - none, run - none, build - none

**Reply by sgkoishi** [reply]
I'm 100% sure [the link above](https://github.com/Pryaxis/TShock/discussions/2911#discussioncomment-4850370) already contains info about "csproj not found".

The dev version contains an installer; screenshot if it's not working

**Reply by jelebruh** [reply]
I forgot about recursive mod ,sry, thanks
